### PR TITLE
Refactored and polished the performance page

### PR DESCRIPTION
**Pull request by nathan-contino**
- Prioritizes high-impact optimizations (such as those discussed with @adammurdoch [here](https://github.com/gradle/gradle/pull/21350#discussion_r930449364)) first on the updated performance page.
- Rephrases titles to make the right sidebar easier to parse and to make sections easier to skip when irrelevant to a reader.
- Rewrites the entire performance page in a consistent voice.
- Replaces many passive sentences with active sentences to better engage readers.
- Splits out "Build scans" details as a separate page.
- Replaces a text-only image on the build scan page with an accessibility-friendly code snippet.
- Trims, polishes, rephrases, rewords, and otherwise optimizes the language of a high-visibility docs page.

### Context
Why waste time, say lot word, when few word do trick?

[Current page](https://docs.gradle.org/current/userguide/performance.html)
[Updated page](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/55055420:id/distributions/gradle-7.7-docs.zip!/gradle-7.7-20220817162105%2B0000/docs/userguide/performance.html)
and [now-separate build scans page](https://builds.gradle.org/repository/download/Gradle_Master_Check_BuildDistributions/55055420:id/distributions/gradle-7.7-docs.zip!/gradle-7.7-20220817162105%2B0000/docs/userguide/inspect.html)

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [X] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [X] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [X] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [X] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by big-guy** [comment]
I haven't had a chance to read the nitty gritty details, but the overall restructuring looks good.

A few thoughts...
- All of these build scan images are woefully out of date. I wouldn't block your PR on updating them, but we should update them at some point.
- In the "Improve the performance" chapter, maybe we should move anything that's already the default (Gradle daemon and incremental compilation) to a different "Improving the performance of older Gradle releases" or something. 
- The "Using Build Scans" chapter looks a little mixed up and sparse. It's really talking about 3 different things (build scans, gradle-profiler and --profile) that are all related to "profiling". 

I wonder if could reuse the title "Inspecting Gradle Builds" (which links to scans.gradle.com) for "Using Build Scans" and make the build scans content more visible. So it would go:

- Build scan content (intro + "profiling with build scans")
- Describe --profile as a more basic local tool (profile report)
- Describe gradle-profiler as a different (and deeper) analysis (low level profiling)

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@big-guy , thanks for the suggestions -- I've incorporated those now. Would appreciate another look when you get the chance.

**Review by adammurdoch — COMMENTED** [review]
I've added a bunch of comments. I wouldn't consider any of them blockers, rather further improvements we could make later.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 1]
```diff
@@ -1,180 +1,250 @@
...
 
-All this means that it’s worth investing some time and effort into making your build as fast as possible.
-This section offers several avenues you can explore to make a build faster, along with plenty of detail on what sorts of things can degrade the performance of a build and why.
+The Gradle team continuously improves the performance of Gradle builds.
+If you’re using an old version of Gradle, you’re missing out on the benefits of that work.
+Keeping up with Gradle version upgrades is low risk because the Gradle team ensures backwards compatibility between minor versions of Gradle.
+Staying up-to-date also makes transitioning to the next major version easier, since you'll get early deprecation warnings.
 
-== Build scans
+== Upgrade to the latest Java version
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc:23]: Mentioning Java as well in this section.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 2]
```diff
@@ -1,180 +1,250 @@
...
-image::performance/parallel-task-fast.png[title="Optimized parallel execution"]
+The configuration cache enables additional optimizations as well. When enabled, Gradle:
+
+- Executes all tasks in parallel, even those in the same subproject.
+- Caches dependency resolution results.
 
-That’s the end of the quick wins.
-From here on out, improving your build performance will require some elbow grease.
-First, perhaps the most important step: finding out which bits of your build are slow and why.
+== Enable incremental build
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc:120]: I like the idea of a "things Gradle does out of the box to make your life easier" section, but I think my edits obfuscated the point of this section: making custom tasks compatible with incremental build.

I've updated the section to include an example specifying inputs and outputs for a custom task. Let me know if you think this makes the section less... nonsensical.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 3]
```diff
@@ -1,180 +1,250 @@
...
+Sort by task duration on the timeline screen to highlight tasks with great time saving potential.
+The build scan above shows that `:task1` and `:task3` could be improved and made cacheable
+and shows why Gradle didn't cache them.
 
-This script takes 1 second to run, and as it is applied to 3 projects from the root build script, it is introducing a 3 second delay to the configuration phase.
+== Configuration
+
+As described in <<build_lifecycle.adoc#build_lifecycle,the build lifecycle chapter>>, a
+Gradle build goes through 3 phases: initialization, configuration, and execution.
+Configuration code always executes regardless of the tasks that run.
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc:260]: Thanks, added a note explaining how the configuration cache impacts this.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 4]
```diff
@@ -565,34 +570,39 @@ The Gradle Java plugin allows you to run the compiler as a separate process by u
...
+Gradle reuses this process within the duration the build, so the forking overhead is minimal.
+By forking memory-intensive compilation into a separate process, we minimize garbage collection in the main Gradle process.
+Less garbage collection means that Gradle's infrastructure can run faster, especially when you also use <<parallel_execution, parallel builds>>.
 
-It's unlikely to be useful for small projects, but you should definitely consider it if a single task is compiling close to a thousand or more source files together.
+Forking compilation rarely impacts the performance of small projects.
+But you should consider it if a single task compiles more than a thousand source files together.
 
-=== Compile avoidance
+==== Compile avoidance
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc]: Moved to the "Older gradle versions" section.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 5]
```diff
@@ -1,180 +1,250 @@
...
+----
+org.gradle.unsafe.configuration-cache=true
+----
 
-image::performance/parallel-task-slow.png[title="Bottleneck in parallel execution"]
+For more information about the configuration cache, check out the
+<<configuration_cache.adoc#config_cache,configuration cache documentation>>.
 
-Tweaking the build configuration to run the two slow tasks early on and in parallel reduces the overall build time from 8 seconds down to 5 seconds:
+=== Additional configuration cache optimizations
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc]: Good point, reworded to "benefits" in title and text.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 6]
```diff
@@ -1,180 +1,250 @@
...
 
-https://gradle.com/build-scans[Build scans] are a persistent, shareable record of what happened when running a build.
-With build scans, you gain deep insights about your build to identify and fix performance bottlenecks.
+Gradle runs on the Java Virtual Machine (JVM). Java performance improvements often benefit Gradle.
+For the best Gradle performance, use the latest version of Java.
 
-If you are using Gradle 4.3+, you can easily create a build scan by using the `--scan` command line option, e.g. `gradle build --scan`.
-For older Gradle versions, see the https://docs.gradle.com/build-scan-plugin/#getting_set_up[Build Scan Plugin User Manual] on how to enable build scans.
+[[parallel_execution]]
+== Enable parallel execution
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc:55]: Introduced a new "enable the daemon" section early on the page.

File: subprojects/docs/src/docs/userguide/running-builds/performance.adoc [hunk 7]
```diff
@@ -1,180 +1,250 @@
...
+The longer builds take to complete, the more likely they'll disrupt your development flow.
+Builds run many times a day, so even small waiting periods add up.
+The same is true for Continuous Integration (CI) builds: the less time they take, the faster you can react to new issues
+and the more often you can experiment.
+
+All this means that it’s worth investing some time and effort into making your build as fast as possible.
+This section offers several ways to make a build faster. Additionally, you'll find details about what leads to
+build performance degradation, and how you can avoid it.
 
 TIP: Want faster Gradle Builds? https://gradle.com/training/build-cache-deep-dive/?bid=guides-performance[Register here] for our Build Cache training session to learn how Gradle Enterprise can speed up builds by up to 90%.
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/performance.adoc:14]: Added an "inspect your build" section based on these suggestions early on the page.

File: subprojects/docs/src/docs/userguide/running-builds/build_scans.adoc [hunk 8]
```diff
@@ -0,0 +1,130 @@
+[[build_scans]]
+= Inspecting Gradle Builds
```
[the remark this answers is not in the document]
Comment (nathan-contino) [reply subprojects/docs/src/docs/userguide/running-builds/build_scans.adoc]: I don't want to bloat the page with too much explanation, but if you feel like this page still needs an introduction, I'm happy to add one. I could say something like:

```
You can inspect your Gradle build in several ways:

- Profile with build scans
- Local profile reports
- Low level profiling
```

Would that be adequate?

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

**Comment by nathan-contino** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/userguide/running-builds/build_scans.adoc (hunk 8 above)
[the remark this answers is not in the document]
Comment (adammurdoch) [reply subprojects/docs/src/docs/userguide/running-builds/build_scans.adoc]: that's exactly what I was thinking.

**Review by adammurdoch — APPROVED** [review]
Looks good.

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Comment by nathan-contino** [comment]
@bot-gradle test and merge

**Pull request merged**